### ISSUE TITLE
[Bug]: "Manifest key not supported by the specified minimum Firefox for Android version" despite not even having set it to Android

### ISSUE THREAD

**Issue by D4n2021** [mozilla/addons-linter#6001]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What happened?

This error, despite me not even using the "browser_specific_settings.gecko.data_collection_permissions" property at all.
(Manifest v2 addon, only "Firefox" selected, "Firefox for Android" NOT selected)

```
...
"browser_specific_settings": {
    "gecko": {
      "id": "{...[redacted]...}",
      "strict_min_version": "140.0",
      "data_collection_permissions": {
        "required": ["none"]
      }
```

<img width="1007" height="192" alt="Image" src="https://github.com/user-attachments/assets/74bc7d3d-fbf4-4b9f-a0bc-9110fc3fea18" />

### What did you expect to happen?

No warning like this at all. No warning should appear at all in this scenario, as I understand

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-2428)

**Comment by willdurand** [comment]
When you submit a new file to AMO, the linter must assume it _can_ target both platforms (Desktop and Android) because the platform selector is shown after this validation step. That's why those are warnings and not errors. Unfortunately, as long as AMO will allow compatibility to be set in DevHub (and not just in the manifest file), we cannot change that.

**Comment by D4n2021** [comment]
Okay! Thx for the explanation
